app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,20 @@
+class LikesController < ApplicationController
+  def new
+    @like = Like.new
+    @user = current_user
+  end
+
+  def create
+    @post = Post.find(params[:post_id] || params[:id])
+    @like = Like.new(post_id: @post.id, author_id: current_user.id)
+    @like.author_id = current_user.id
+
+    respond_to do |format|
+      if @like.save
+        format.html { redirect_to user_post_url(current_user, @post), notice: 'Like was successfully created.' }
+      else
+        format.html { render :new, status: :unprocessable_entity }
+      end
+    end
+  end
+end
```

Comment:
OPTIONAL

- [ ] When clicking on the add-like button ıt it opens a new page that has another add-like button and after ı click on the button, post liked. No need to open a new page for just like. You can fix it.

![image](https://user-images.githubusercontent.com/38635158/181918577-20c93e0d-b651-4407-8b36-a0a6f2fa6165.png)